internal/confmapprovider/discovery/README.md

```diff
@@ -101,4 +101,50 @@ successfully started observers.
 1. Stop all temporary components before continuing on to the actual Collector service (or exiting early with `--dry-run`).
 
 
-By default, the Discovery mode is provided with pre-made discovery config components in [`bundle.d`](./bundle/README.md).
\ No newline at end of file
+By default, the Discovery mode is provided with pre-made discovery config components in [`bundle.d`](./bundle/README.md).
+
+
+### Discovery properties
+
+Configuring discovery components is performed by merging discovery properties with the config.d receivers
+and extensions `*.discovery.yaml` files. Discovery properties are of the form:
+
+```yaml
+splunk.discovery.receivers.<receiver-type(/name)>.config.<field>(<::subfield>)*: <value>
+splunk.discovery.extensions.<observer-type(/name)>.config.<field>(<::subfield>)*: <value>
+splunk.discovery.receivers.<receiver-type(/name)>.enabled.: <true or false>
+splunk.discovery.extensions.<observer-type(/name)>.enabled: <true or false>
+
+# Examples
+splunk.discovery.receivers.prometheus_simple.config.labels::my_label: my_label_value
+splunk.discovery.receivers.prometheus_simple.enabled: true
+
+splunk.discovery.extensions.docker_observer.config.endpoint: tcp://localhost:8080
+splunk.discovery.extensions.k8s_observer.enabled: false
+```
+
+These properties can be in `config.d/properties.discovery.yaml` or specified at run time with `--set` command line options.
+
+Each discovery property also has an equivalent environment variable form using `_x<hex pair>_` encoded delimiters for
+non-word characters `[^a-zA-Z0-9_]`:
```

Comment:
Could you maybe provide an specific example of the mapping from the yaml format to the equivalent environment variable form? It took me quite a while to understand how exactly this mapping worked.

Reply:
The guidance for individual receivers will include their form for c/p and I'm still thinking about how to expose the helper w/ a cli: https://github.com/signalfx/splunk-otel-collector/blob/cb4e27e2dc4010d93ed3e8abbe1d5e9e790ace7e/internal/confmapprovider/discovery/bundle/templatefunctions.go#L33